app/services/workflow_link_service.rb

```diff
@@ -0,0 +1,28 @@
+class WorkflowLinkService
+  attr_accessor :workflow_id
+
+  def initialize(workflow_id)
+    self.workflow_id = workflow_id
+  end
+
+  def link(options)
+    TagLink.where(tag_link(options)).first_or_create
+    nil
+  end
+
+  def unlink(options)
+    TagLink.find_by(tag_link(options)).try(:destroy)
+  end
+
+  def find(options_arr)
+    options_arr.collect do |options|
+      TagLink.where(tag_link(options)).pluck(:workflow_id).first
```

Comment:
@bzwei Is this returning an array of workflow ids? How would we know which array entry corresponds to which workflow.

Reply:
This is like an array transform method many languages support: the output array has the same  length of the input, one to one mapping.